packages/libs/eda/src/lib/core/components/computations/plugins/correlationAssayAssay.tsx

```diff
@@ -198,3 +204,17 @@ export function CorrelationAssayAssayConfiguration(
     </ComputationStepContainer>
   );
 }
+
+// The correlation assay x assay app should only be available
+// for studies with metagenomic data.
+function isEnabledInPicker({
+  studyMetadata,
+}: IsEnabledInPickerParams): boolean {
+  if (!studyMetadata) return false;
+
+  const entities = entityTreeToArray(studyMetadata.rootEntity);
+  const hasMetagenomicData =
+    entities.filter((entity) => entity.id === 'OBI_0002623').length > 0; // OBI_0002623 = Metagenomic sequencing assay
+
```

Comment:
you're right we should. I was making the assumption that if that entity existed, then it would have a collection. But just becasue that's generally true doesn't mean it will always be the case! I'll add a check

Reply:
Also, an `entities.find(...) !== undefined` might be very very slightly marginally more efficient! (because it stops the search as soon as it finds something - but these arrays are tiny, so it will make very little difference)

Reply:
Or I guess `entities.some(...)` works too!

Reply:
It's still possible that the study with collections and a OBI_0002623 assay with no collections could get through, I think?

Maybe just this?

```
  const hasMetagenomicData = entities.some((entity) => entity.id === 'OBI_0002623' && !!entity.collections?.length);
```

Reply:
ack you're right! All fixed now!

I was interested so I looked into the difference in performance between `some` and `find`. It seems `some` is ever so slightly more efficient, since `find` calls `some`. But you're right it wouldn't matter on an array of this size anyway.